db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLTimestamp.java

```diff
@@ -1584,6 +1586,49 @@ public void write(UnsafeRowWriter unsafeRowWriter, int ordinal) {
 				holder.cursor = 12;
 			}
 	    }
+
+	/**
+	 *
+	 * Write Element into Positioned Array
+	 *
+	 * @param unsafeArrayWriter
+	 * @param ordinal
+	 * @throws StandardException
+     */
+	@Override
+	public void writeArray(UnsafeArrayWriter unsafeArrayWriter, int ordinal) throws StandardException {
+		if (isNull())
+			unsafeArrayWriter.setNull(ordinal);
+		else {
+			byte[] data = new byte[12];
```

Comment:
Platform API's are a bit different.  They go against the actual memory.  The first 16 bytes of memory is held for the array object instance and references in java.